src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/DeleteDataSetVersionFunction.cs

```diff
@@ -0,0 +1,31 @@
+using GovUk.Education.ExploreEducationStatistics.Common.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Services.Interfaces;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Extensions.Logging;
+
+namespace GovUk.Education.ExploreEducationStatistics.Public.Data.Processor.Functions;
+
+public class DeleteDataSetVersionFunction(IDataSetVersionService dataSetVersionService, ILogger<DeleteDataSetVersionFunction> logger)
+{
+    [Function(nameof(DeleteDataSetVersion))]
+    public async Task<IActionResult> DeleteDataSetVersion(
+        [HttpTrigger(AuthorizationLevel.Function, "delete", Route = $"{nameof(DeleteDataSetVersion)}/{{dataSetVersionId}}")] HttpRequest httpRequest,
```

Comment:
It doesn't appear like it unfortunately. I get a binding error when I don't specify a `HttpRequest httpRequest` parameter. It also needs to be named `httpRequest` exactly. Looks like a weird quirk built into the framework.

Reply:
Ok cool, that's disappointing.

One post I saw suggested using `string dataSetVersionId` and then parsing it to Guid later might work, but it doesn't seem to unfortunately.

While I was trying it out myself I found that it doesn't have to be named `httpRequest` as in my test case I was using `HttpRequest request` and this was ok. This makes sense because our `HealthCheckFunctions.HealthCheck` is already using `HttpRequest request` too.

One thing you might want to change though is constraining the type of the `dataSetVersionId` parameter to Guid using

```csharp
[HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route=${nameof(DeleteDataSetVersion)}/{{dataSetVersionId:guid}}")]
```

If you do this and the value is not a valid Guid then it doesn't match the route.

Reply:
Ah yeah, now that you mention it I vaguely remember reading that there are a few allowed names for that variable. `request` may have been one of them.

I'll change it to `request` and I'll add the `:guid` constraint, nice spot!